server/tools/inventory-tools.ts

```diff
@@ -0,0 +1,425 @@
+        DestinyComponentType.ItemStats, // All item stats
+        DestinyComponentType.ItemPerks, // Intrinsic perks
+      ],
+    });
+
+    if (!profileResponse?.Response) {
+      throw new Error('Failed to fetch profile from Bungie API');
+    }
+
+    // Collect all raw items from profile
+    const profile = profileResponse.Response;
+    const allItems: any[] = [];
+
+    // Add vault items
+    if (profile.profileInventory?.data?.items) {
+      for (const item of profile.profileInventory.data.items) {
+        allItems.push({
+          ...item,
+          location: 'vault',
+          equipped: false,
+        });
+      }
+    }
+
+    // Add character items
+    if (profile.characterInventories?.data) {
+      for (const [characterId, inventory] of Object.entries(profile.characterInventories.data)) {
+        for (const item of inventory.items) {
+          allItems.push({
+            ...item,
+            location: 'character',
+            characterId,
+            equipped: false,
+          });
+        }
+      }
+    }
+
+    // Add equipped items
+    if (profile.characterEquipment?.data) {
+      for (const [characterId, equipment] of Object.entries(profile.characterEquipment.data)) {
+        for (const item of equipment.items) {
+          allItems.push({
+            ...item,
+            location: 'character',
+            characterId,
+            equipped: true,
+          });
+        }
+      }
+    }
+
+    console.log(`[MCP] Collected ${allItems.length} items from profile`);
+
+    // Filter to only weapons and armor
+    const weaponsAndArmor = allItems.filter((item) => {
+      const itemDef = itemDefs[item.itemHash];
+      return (
+        itemDef?.itemType === DestinyItemType.Weapon ||
+        itemDef?.itemType === DestinyItemType.Armor
+      );
+    });
+
+    console.log(`[MCP] Filtered to ${weaponsAndArmor.length} weapons and armor`);
+
+    // Enrich items with their definitions and instance data
+    // Note: Weapons and armor are always instanced, so we can safely assume itemInstanceId exists
+    const enrichedItems = weaponsAndArmor.map((item) => {
+      const itemDef = itemDefs[item.itemHash];
+      const statsData = profile.itemComponents?.stats?.data?.[item.itemInstanceId];
+      const perksData = profile.itemComponents?.perks?.data?.[item.itemInstanceId];
+
+      // Map stats with names - filter to only important stats
+      const importantStats = [
+        'Stability',
+        'Handling',
+        'Range',
+        'Reload Speed',
+        'Impact',
+        'Magazine',
+        'Rounds Per Minute',
+        'Mobility',
+        'Resilience',
+        'Recovery',
+        'Discipline',
+        'Intellect',
+        'Strength',
+      ];
+
+      // Get stats from investmentStats (like DIM does internally)
+      // This is more reliable than the ItemStats component, especially for armor
+      let enrichedStats: Array<{name: string; value: number}> = [];
```

Comment:
I need to keep iterating on this and am open to feedback. I couldn't get stats to work for Armors